modules/programs/procmail.nix

```diff
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.procmail;
+  ruleModule = types.submodule ({config,...}: {
+    options = {
+
+      lock = mkOption {
+        type = types.string;
+        default = ":0:";
+        description = "Locking rule.";
+      };
+
+      conditions = mkOption {
+        type = types.listOf types.string;
+        default = [];
+        example = ["^From: .*@linkedin.com"];
+        description = "Matching rules.";
+      };
+
+      target = mkOption {
+        type = types.string;
+        default = "";
+        example = "mbox/";
+        description = "Rule target. If you want a maildir, append a slash.";
+      };
+    };
+  });
+
+in
+
+{
+  options = {
+    programs.procmail = {
+      enable = mkEnableOption "the procmail mail processor";
+
+      maildir = mkOption {
+        type = types.string;
+        default = "$HOME/Mail";
+        description = "Base mail directory.";
+      };
+
+      finalDir = mkOption {
+        type = types.string;
+        default = "mbox/";
+        description = "The final catch-all maildir.";
+      };
+
+      # Still thinking of proper syntax for this
+      preConfig = mkOption {
+        type = types.lines;
+        default = "";
+        example = ''
+          # START spamassassin
+          :0fw: spamassassin.lock
+          * < 256000
+          | ${pkgs.spamassassin}/bin/spamc
+        '';
+        description = "Configuration to add to the top after the header.";
+      };
+
+      rules = mkOption {
+        type = types.listOf ruleModule;
+        default = [];
+        example = [
+          {
+            rules = ["^X-BeenThere: haskell-cafe@haskell.org"];
+            target = "~/Mail/Lists/haskell-cafe/";
+          }
+        ];
+        description = "List of procmail rules.";
+      };
+
+      postConfig = mkOption {
+        type = types.lines;
```

Comment:
traditionnally, it is called extraConfig. It would be best to have some kind of consistency there.

Reply:
`postConfig` vs `preConfig`, I wanted to have a separate section that comes before and after the rules. For example spamassassin filtering doesn't fit the rules format, and needs to come before the rules.

Reply:
my bad I missed the preConfig part. Calling these pre/post ExtraConfig may make them more discoverable (though it's fine as it is too).